SU-SWS/ansible-playbooks
# SITES-319: Draft of flagging php in database

# READY FOR REVIEW

# Summary
- We want to find and abort the migration of sites with PHP in the database.  If there is a standard way of parsing data for PHP, please let me know (this has proven a difficult think to search for as it returns parsing functions in PHP).  Based on a small subset of test databases, the following regex seemed to do a pretty good job of flagging PHP, while excluding non sequiturs.  But we will likely need to add more over time.

```
    - '.\{0,20\}$[^S$D|\.|0-9|conf|user|_SESSION][A-Za-z].\{0,20\}'
    - '.\{0,20\}<?[^xml].\{0,20\}'
    - 'eval('
    - 'system('
    - 'preg_replace('
    - 'create_function'
    - 'include\['
    - 'require\['
```

# Needed By (2/1)
- End of sprint

# Urgency
- While important, not urgent.

# Steps to Test

1. I've been testing with SITES224 and eao.  SITES224 should log the following lines:

```
        "lay_format\\\";s:46:\\\"$display_output = $enti",
        "lay_format\\\";s:46:\\\"$display_output = $enti",
        "lay_format\\\";s:46:\\\"$display_output = $enti",
        "lay_format\\\";s:46:\\\"$display_output = $enti"
```

2. Only the eao host should move forward with the migration.
3. Check that the PHP above is logged in /tmp/ansible.log.

# Affected Projects or Products
- Sites 2.0 security efforts

# Associated Issues and/or People
- https://stanfordits.atlassian.net/browse/SITES-319
- Thanks to Dmitry for helping me figure out how to log the grep results before I fail and abandon the host with PHP in the database.

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
